brokerapi/brokers/spanner/broker.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	// save off instance information
+	ii := InstanceInformation{
+		InstanceId: params["name"],
+	}
+
+	otherDetails, err := json.Marshal(ii)
+	if err != nil {
+		return models.ServiceInstanceDetails{}, fmt.Errorf("Error marshalling other details: %s", err)
+	}
+
+	i := models.ServiceInstanceDetails{
+		Name:         params["name"],
+		Url:          "",
+		Location:     loc,
+		OtherDetails: string(otherDetails),
+	}
+
+	//op := client.InstanceOperation(params["name"])
+	err = createCloudOperation(op, instanceId, details.ServiceID)
+	if err != nil {
+		return models.ServiceInstanceDetails{}, fmt.Errorf("Error saving operation to database: %s", err)
+	}
+
+	return i, nil
+}
+
+// gets the last operation for this instance and polls the status of it
+func (s *SpannerBroker) PollInstance(instanceId string) (bool, error) {
+
+	var op models.CloudOperation
+
+	if err := db_service.DbConnection.Where("service_instance_id = ?", instanceId).First(&op).Error; err != nil {
+		return false, fmt.Errorf("Could not locate CloudOperation in database")
+	}
+
+	var instance models.ServiceInstanceDetails
+
+	if err := db_service.DbConnection.Where("id = ?", instanceId).First(&instance).Error; err != nil {
+		return false, models.ErrInstanceDoesNotExist
+	}
+
+	client, err := googlespanner.NewInstanceAdminClient(context.Background())
+	if err != nil {
+		return false, fmt.Errorf("Error creating client: %s", err)
+	}
+
+	spannerOp := client.CreateInstanceOperation(op.Name)
+
+	spannerInstance, err := spannerOp.Poll(context.Background())
+	done := spannerOp.Done()
+
+	// from https://godoc.org/cloud.google.com/go/spanner/admin/instance/apiv1#InstanceOperation.Poll
+	if spannerInstance == nil && err != nil && !done {
+		return false, fmt.Errorf("Error checking operation status: %s", err)
+	} else if spannerInstance == nil && err != nil && done {
+		op.Status = "FAILED"
+		op.ErrorMessage = err.Error()
+
+		if err = db_service.DbConnection.Save(&op).Error; err != nil {
+			return false, fmt.Errorf(`Error saving operation details to database: %s.`, err)
+		}
+
+		return true, fmt.Errorf("Error provisioning instance: %s", err)
+	} else if spannerInstance == nil && err == nil && !done {
+		op.Status = spannerInstance.State.String()
+
+		if err = db_service.DbConnection.Save(&op).Error; err != nil {
+			return false, fmt.Errorf(`Error saving operation details to database: %s.`, err)
+		}
+
+		return false, nil
+	} else if spannerInstance != nil && err == nil && done {
+		op.Status = spannerInstance.State.String()
+
+		if err = db_service.DbConnection.Save(&op).Error; err != nil {
+			return false, fmt.Errorf(`Error saving operation details to database: %s.`, err)
+		}
+
+		return true, nil
+	}
+
+	return false, fmt.Errorf("unknown error")
+}
+
+func createCloudOperation(op *googlespanner.CreateInstanceOperation, instanceId string, serviceId string) error {
+	var err error
+
+	_, err = op.Poll(context.Background())
+
+	errorStr := ""
+	if err != nil {
```

Comment:
another way to do this, since you don't care about the first return of `op.Poll`:

```go
errorStr := ""
if _. err := op.Poll(context.background(); err != nil {
   errorStr := err.Error()
}
```

Then no var error and it's a bit more compact.